### PR TITLE
hero-18-hats-protocol-publish-a-cast-from-a-farcaster-account-that

### LINKED ISSUE #110: Hats Protocol - Publish a cast from a Farcaster account that is managed via Hats Protocol
**Assumptions**:
- existing Farcaster account setup via non-herocast tool

### PR DESCRIPTION
**Pull request by hellno**
wip
fixes #110

### CODE REVIEW

File: src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+        key: hexStringPublicKey,
+        deadline: BigInt(deadline),
+      },
+    };
+    const metadataHash = hashTypedData(typedMetadataData);
+    const metadataSignature = await signTypedDataAsync(typedMetadataData);
+    console.log('metadataSignature', metadataSignature)
+
+    const METADATA_TYPEHASH = '0x16be47f1f1f50a66a48db64eba3fd35c21439c23622e513aab5b902018aec438';
+    const metadata = encodePacked(
```
Comment (spengrah) [inline src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx:338]: this might need to be `encode()` rather than `encodePacked()`

Comment (spengrah) [reply src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx:338]: more specifically, I think the last four values should be concatenated with `encode()`, and the result should be `encodePacked()` with the `metadataSignature`.

File: src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+        keyType: 1, // only supports 1 as key type
+        key: hexStringPublicKey,
+        metadataType: 1, // only supports 1 as metadata type
+        metadata,
+        deadline,
+        nonce,
+      },
+    };
+
+    const addHash = hashTypedData(typedData);
```
Comment (spengrah) [inline src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx:362]: this is using the hashedTypedData function from the Key Gateway, right?

File: src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+    const addSignature = await signTypedDataAsync(typedData);
+
+    // signature, typehash, and then all the args
+    const sig = encodePacked(
```
Comment (spengrah) [inline src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx:366]: same here as with the `metadata` encoding: the last 8 values should be `encode()`-concatenated together, and the result should be `encodePacked()` with the `addSignature`

File: src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx (hunk 2 above)
Comment (hellno) [reply src/common/components/HatsProtocol/ConnectFarcasterAccountViaHatsProtocol.tsx:362]: yes

**Pull request merged**